compute/managed-instances/demo/main.py

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/python3
+from http.server import BaseHTTPRequestHandler, HTTPServer
+from jinja2 import Environment, FileSystemLoader, select_autoescape
+from os import path
+from requests import get
+from re import sub
+from socket import gethostname
+from socketserver import ThreadingMixIn
+
+PORT_NUMBER = 80
+HTML_TEMPLATES_PATH = path.dirname(path.realpath(__file__)) + '/templates'
+HEALTHY = True
+
+env = Environment(
+    loader=FileSystemLoader(HTML_TEMPLATES_PATH),
+    autoescape=select_autoescape('html')
+)
+
+
+class request_handler(BaseHTTPRequestHandler):
+    def get_zone(self):
```

Comment:
These helper functions don't actually use "self" passed in, so perhaps they should be top-level standalone functions instead of instance methods on this class. I was confused when I read them because I expected that since they are instance methods of a request handler, they would handle requests and return HTTP responses.

Reply:
Refactored with update to Flask